scripts/stub-completed-container-builds.py

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0+
+
+import argparse
+import datetime
+from neomodel import db
+
+from assayist.common.models.content import Build
+from assayist.processor.configuration import config
+from assayist.processor.utils import get_koji_session, is_container_build
+
+
+def batch_dates_for_range(start_date, end_date):
+    """
+    Generate batches of dates that cover the given time period.
+
+    Koji runs out of memory if we ask about a whole year's worth of builds at a time, so
+    let's batch the requests up.
+
+    :param Date start_date: First start date to return.
+    :param Date end_date: Date to continue until, inclusive.
+    :return: An interable of tuples of ISO-formatted string datetimes, start and end respectively.
+    :rtype: [(string, string)]
+    """
+    period = datetime.timedelta(days=10)
+    end_date = end_date + datetime.timedelta(days=1)  # We want to find builds done on end_date too.
+
+    def dt_string(x):
+        return str(x) + ' 00:00:00'
+
+    while (start_date + period) < end_date:
+        yield dt_string(start_date), dt_string(start_date + period)
+        start_date = start_date + period
+
+    yield dt_string(start_date), dt_string(end_date)
+
+
+def valid_date(s):
+    """
+    Convert input string to a date, or error.
+
+    https://stackoverflow.com/questions/25470844/specify-format-for-input-arguments-argparse-python
+    """
+    try:
+        return datetime.datetime.strptime(s, '%Y-%m-%d').date()
```

Comment:
You can return the whole datetime here (instead of just the date) since later on you're appending `00:00:00` to it anyway. Then you can get rid of the `dt_string()` function and simply return `str(start_date)`.

Reply:
Except then you'd have the opposite problem if `end` is not specified, you'd need to turn a date (`today()`) into a datetime. Or use `now()` or `utcnow()` and figure out what timezone Koji expects things to be in. If the user is inputing dates then I prefer to keep things as simple dates as long as possible.